### ISSUE TITLE
Change default delete action

### ISSUE THREAD

**Issue by ducalex** [dundee/gdu#140]
In gdu the default button in the delete dialog is "Yes" but in ncdu it is "No". This isn't a big deal but muscle memory can be hard to adapt... I would like to add an option to change the default in gdu to match the behaviour of ncdu.

Is this feature something the gdu project would accept to merge, and if so what do you think would be the best way to do it? Flag? Env? Other?

Thanks!

PS: I did implement this feature for myself, but it's not elegant (added --default-delete-action=yes|no but it required changing CreateUI signature to pass the flag which of course ends up changing way too much stuff...).

**Comment by daniejstriata** [comment]
I prefer the `yes` default myself but if it can become a configurable item it would make sense.

**Comment by dundee** [comment]
@ducalex: Yes, it has to bubble through quite a lot of code. Adding it makes sense to me as well.

**Comment by ducalex** [comment]
> @ducalex: Yes, it has to bubble through quite a lot of code. Adding it makes sense to me as well.

Thank you, I'll spend some time studying the code and prepare a draft with hopefully minimal and non disruptive changes!


On a related note, would it make sense to pass a UIConfig object (or similar appropriate construct) to CreateUI instead of a list of args? Because there are other feature requests to add flags/settings (#12/#123, #103) and I suspect it would involve updating ~70 CreateUI calls (which admittedly is mostly tests) for every addition?

**Comment by dundee** [comment]
Yes, passing the options as config object makes sense to me. We can make separate PR for it. Would you like to do it?

**Comment by ducalex** [comment]
I'm fairly new at Go so I don't feel confident enough to architect this change myself at the moment. 
But maybe someone will be able to give me some guidance after seeing the extent of code that had to be changed by my initial PR.

**Comment by namespacebrian** [comment]
I wish I could specify a custom function/script for deletion... so I could script to tarball then delete. ;D

**Comment by urshurak** [comment]
Hello,

I'm writing to report a critical usability and safety issue regarding the 'empty directory' functionality in gdu.

Problem Description:
When running gdu and pressing the lowercase 'e' key, gdu directly presents a confirmation dialog to "empty" the currently selected directory. The most concerning aspect is that the default selected option in this confirmation dialog is 'yes'.

This behavior creates an extremely high risk of accidental data loss. A single, brief, unintended press of the 'e' key followed by a quick press of 'Enter' (or another confirmation) can instantly wipe the contents of a directory.

A friend of mine recently experienced significant data loss (approximately 150GB) directly because of this. He was navigating within gdu and inadvertently pressed 'e', and then confirmed the "empty" action for a large directory, resulting in the immediate loss of their data.

Friend's System Details (where the data loss occurred):

    Operating System: Fedora

    Terminal emulator: Tilix

    Desktop Environment: GNOME

Steps to Reproduce:

    Launch gdu (e.g., gdu or gdu -i /media,/run/timeshift).

    Navigate to any directory.

    Press the lowercase 'e' key.

    Observe the confirmation dialog "Are you sure you want to empty '[directory_name]'?" appearing.

    Crucially, observe that the 'yes' option is pre-selected by default.

Expected Behavior / Proposed Solution:

    For any destructive action (like "empty" or "delete"), the default selected option in the confirmation dialog should always be 'no' to prevent accidental data loss. This is a standard safety best practice in user interface design.

    If the intention is for 'e' to be the "empty" key, this should be explicitly documented and perhaps a more distinct keybinding chosen to reduce accidental presses, or at minimum, the default confirmation should be 'no'.

Screenshots:
I have attached a screenshot illustrating the unexpected dialog appearing when pressing 'e' for a directory, clearly showing the 'yes' option as default. (Here you would drag and drop your screenshots into the comment box on GitHub).

![Image](https://github.com/user-attachments/assets/71f4dc0b-c3c1-47df-b5e2-4784866028f8)

This issue presents a serious risk of data loss for gdu users. I hope this detailed report helps in addressing this critical concern promptly.

Thank you for your work on gdu.

Best regards,

**Comment by ShivamB25** [comment]
@dundee please close this down

**Comment by ShivamB25** [comment]
@urshurak done and fixed

**Comment by dundee** [comment]
Will be released soon.
